module/VuFind/src/VuFind/ILS/Driver/SierraRest.php

```diff
@@ -429,6 +436,21 @@ public function setConfig($config)
      */
     public function init()
     {
+        if (empty($this->config)) {
+            throw new ILSException('Configuration needs to be set.');
+        }
+        if ($this->config['InnReach']['enabled'] ?? false) {
+            try {
+                $conn_string = 'host=' . $this->config['InnReach']['dna_url']
+                    . ' port=' . $this->config['InnReach']['dna_port']
+                    . ' dbname=' . $this->config['InnReach']['dna_db']
+                    . ' user=' . $this->config['InnReach']['dna_user']
+                    . ' password=' . $this->config['InnReach']['dna_password'];
+                $this->db = pg_connect($conn_string);
```

Comment:
Since the connection is needed in only certain situations, it'd be better to defer opening it before it's actually needed. Add something like `protected method getInnReachDb()` that initialized the member variable as necessary and returns it. I'd also rename `$this->db` to `$this->innReachDb` to make it clear that it's not used for Sierra.

Also the return value should be checked and any errors handled appropriately (at least errors should be logged, see [pg_last_error](https://www.php.net/manual/en/function.pg-last-error.php)). I'm not sure whether that means logging the error and continuing without INN-Reach support, or giving up and throwing an exception.